community/consistency-check/src/main/java/org/neo4j/consistency/checking/full/ConsistencyCheckTasks.java

```diff
@@ -53,120 +60,166 @@
-                           StoreProcessor processor )
+    private final StoreProcessor defaultProcessor;
+    private final StoreAccess nativeStores;
+    private final Statistics statistics;
+    private final MultiPassStore.Factory multiPass;
+    private final ConsistencyReporter reporter;
+    private final LabelScanStore labelScanStore;
+    private final IndexAccessors indexes;
+    private final CacheAccess cacheAccess;
+    private final int numberOfThreads;
+
+    ConsistencyCheckTasks( ProgressMonitorFactory.MultiPartBuilder progress,
+            StoreProcessor defaultProcessor, StoreAccess nativeStores, Statistics statistics,
+            CacheAccess cacheAccess, LabelScanStore labelScanStore,
+            IndexAccessors indexes, MultiPassStore.Factory multiPass, ConsistencyReporter reporter, int numberOfThreads )
     {
         this.progress = progress;
-        this.order = order;
-        this.processor = processor;
+        this.defaultProcessor = defaultProcessor;
+        this.nativeStores = nativeStores;
+        this.statistics = statistics;
+        this.cacheAccess = cacheAccess;
+        this.multiPass = multiPass;
+        this.reporter = reporter;
+        this.labelScanStore = labelScanStore;
+        this.indexes = indexes;
+        this.numberOfThreads = numberOfThreads;
     }
 
-    public List<StoppableRunnable> createTasks(
-            StoreAccess nativeStores, LabelScanStore labelScanStore, IndexAccessors indexes,
-            MultiPassStore.Factory multiPass, ConsistencyReporter reporter,
-            boolean checkLabelScanStore, boolean checkIndexes, boolean checkGraph )
+    public List<ConsistencyCheckerTask> createTasksForFullCheck( boolean checkLabelScanStore, boolean checkIndexes,
+            boolean checkGraph )
     {
-        List<StoppableRunnable> tasks = new ArrayList<>();
-
+        List<ConsistencyCheckerTask> tasks = new ArrayList<>();
         if ( checkGraph )
         {
-            tasks.add( create( nativeStores.getNodeStore(),
-                    multiPass.processors( LABELS, PROPERTIES, RELATIONSHIPS, RELATIONSHIP_GROUPS ) ) );
-
-            tasks.add( create( nativeStores.getRelationshipStore(),
-                    multiPass.processors(  NODES, PROPERTIES, RELATIONSHIPS  ) ) );
-
-            tasks.add( create( nativeStores.getPropertyStore(),
-                    multiPass.processors(  PROPERTIES, STRINGS, ARRAYS, PROPERTY_KEYS ) ) );
-
-            tasks.add( create( nativeStores.getStringStore(), multiPass.processors( STRINGS ) ) );
-
-            tasks.add( create( nativeStores.getArrayStore(), multiPass.processors( ARRAYS ) ) );
-
-            tasks.add( create( nativeStores.getRelationshipGroupStore(), multiPass.processors( RELATIONSHIP_GROUPS ) ) );
+            MandatoryProperties mandatoryProperties = new MandatoryProperties( nativeStores );
+            StoreProcessor processor =
+                    multiPass.processor( CheckStage.Stage1_NS_PropsLabels, PROPERTIES );
```

Comment:
looks like if we will introduce stages hierarchy it should be much more easier to construct required tasks, wdyt?